### PR TITLE
Lint tweaks

### LINKED ISSUE #198: bin/rtlcss.js fails in strict mode
I have a WIP branch to modernize the codebase and switch to strict mode, but apparently bin/rtlcss.js fails in strict mode

https://github.com/XhmikosR/rtlcss/actions?query=branch%3Alint

### PR DESCRIPTION
**Pull request by XhmikosR**
I will try to split this in the next days.

Fixes #198 along the way.

### CODE REVIEW

**Comment by MohammadYounes** [comment]
Not sure if this is worth the effort at the moment, I was planning on rewriting the whole thing in TS. However, due to limited availability I didn't go through with it.

**Comment by XhmikosR** [comment]
The strict mode is totally worth it IMHO. The object shorthands etc I can skip them, sure.

I'll rebase the branch and keep only the other changes and see where we are at.

**Comment by XhmikosR** [comment]
@MohammadYounes I think I split this as good as I could. Are any of these patches something you'd like to have? Let me know which ones so that I rebase and skip any others.

File: bin/rtlcss.js [hunk 1]
```diff
@@ -13,18 +14,18 @@ let input, output, directory, ext, config, currentErrorcode, arg
...
 let shouldBreak = false
 
-process.on('exit', function () { process.reallyExit(currentErrorcode) })
+process.on('exit', () => { process.reallyExit(currentErrorcode) })
 
-function printWarning () {
-  console.warn(chalk.yellow.apply(this, printWarning.arguments))
+function printWarning (args) {
+  console.warn(chalk.yellow(args))
 }
```
Comment (MohammadYounes) [inline bin/rtlcss.js:21]: @XhmikosR These are not equivalent, isn't it better to have them as ?
```js
function printWarning (...args) {
  console.warn(chalk.yellow.apply(this, args))
}
```

Comment (XhmikosR) [reply bin/rtlcss.js:21]: Oops sorry, my bad.

```js
function printWarning (...args) {
  console.warn(chalk.yellow(args))
}
```

The problem is that the previous code does not work in strict mode.

**Comment by MohammadYounes** [comment]
@XhmikosR LGTM, I just need to try out the possibility of having `strict` mode with default `standard`, I'll check it during the weekend.

Thank you!

File: bin/rtlcss.js (hunk 1 above)
Comment (MohammadYounes) [reply bin/rtlcss.js]: Then:

```js
function printWarning (...args) {
  console.warn(chalk.yellow(...args))
}
```

Comment (XhmikosR) [reply bin/rtlcss.js]: Sorry, third time is the charm 😔 Doing too many things at the same time.

**Comment by MohammadYounes** [comment]
@XhmikosR Reverting back to default standard 0eb67c5d2577d0d4f8d16b49b4cf339e2aae43b4 raises no issues!

https://github.com/MohammadYounes/rtlcss/actions/runs/423683876

**Comment by XhmikosR** [comment]
@MohammadYounes yeah, but we can't enforce any custom rules with `standard` hence why I switched it to `standardx` :)

**Comment by MohammadYounes** [comment]
@XhmikosR That's the idea behind using `standard`, keeping it simple without any custom rules :)

**Comment by XhmikosR** [comment]
Yeah, I know but IMHO it lacks many things.

This branch is moot if the new rules aren't enforced. In that case I
suggest that I split 1-2 patches and drop the other.

**Comment by XhmikosR** [comment]
I insist, there's no point in most of these patches if the rules aren't enforced :)

I suggest we only keep these patches then:

* https://github.com/MohammadYounes/rtlcss/pull/199/commits/38849fd669db334323d08ba6281a896ea30e8be6
* https://github.com/MohammadYounes/rtlcss/pull/199/commits/194b3287c402fd2588830757e3997bcabd7e3a24
* https://github.com/MohammadYounes/rtlcss/pull/199/commits/23ffc7d60c403cdf9f2942967c12ada87e3d26d0

Even though guard-for-in isn't enabled in standard, I still think it's good to have it to protect from prototype pollution issues.

**Comment by MohammadYounes** [comment]
@XhmikosR Fine. However, until a full rewrite, I don't expect current codebase to grow further.  So I don't think enforcing the rules now makes any difference :)

**Comment by XhmikosR** [comment]
Rebased and kept the aforementioned patches only. I'm still not sure if it's worth changing the parser options...

**Comment by MohammadYounes** [comment]
@XhmikosR I still see no need to switch to `standardx`, the for-in guard is already validated by github actions.

**Comment by XhmikosR** [comment]
Not in all places, though. Anyway, feel free to drop this branch. It's basically the same as standard but it allows us to extend its config.

**Comment by XhmikosR** [comment]
Alternatively, I can revert the standardx change, but keep in mind that these rules aren't enforced by stabndard.

**Comment by MohammadYounes** [comment]
I prefer to keep the branch (for strict mode) and revert the `standardx` change. Thanks!

**Pull request merged**